plugins/outputs/kairosdb/kairosdb.go

```diff
@@ -0,0 +1,287 @@
+	}
+	connection, err := net.DialTCP("tcp", nil, tcpAddr)
+	if err != nil {
+		return fmt.Errorf("KairosDB Telnet connect fail: %s", err)
+	}
+	defer connection.Close()
+	return nil
+}
+
+func (o *KairosDB) Write(metrics []telegraf.Metric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	u, err := url.Parse(o.Host)
+	if err != nil {
+		return fmt.Errorf("Error in parsing host url: %s", err.Error())
+	}
+
+	if u.Scheme == "" || u.Scheme == "tcp" {
+		return o.WriteTelnet(metrics, u)
+	} else if u.Scheme == "http" || u.Scheme == "https" {
+		return o.WriteHttp(metrics, u)
+	} else {
+		return fmt.Errorf("Unknown scheme in host parameter.")
+	}
+}
+
+func (o *KairosDB) WriteHttp(metrics []telegraf.Metric, u *url.URL) error {
+	http := kairosDBHttp{
+		Host:      u.Host,
+		Port:      o.Port,
+		Username:  o.Username,
+		Password:  o.Password,
+		Scheme:    u.Scheme,
+		User:      u.User,
+		BatchSize: o.HttpBatchSize,
+		Path:      o.HttpPath,
+		Debug:     o.Debug,
+	}
+
+	for _, m := range metrics {
+		now := m.Time().UnixNano() / 1000000
+		tags := cleanTags(m.Tags())
+
+		for fieldName, value := range m.Fields() {
+			switch value.(type) {
+			case int64:
+			case uint64:
+			case float64:
+			default:
+				log.Printf("D! KairosDB does not support metric value: [%s] of type [%T].\n", value, value)
+				continue
+			}
+
+			metric := &HttpMetric{
+				Metric: sanitize(fmt.Sprintf("%s%s%s%s",
+					o.Prefix, m.Name(), o.Separator, fieldName)),
+				Tags:      tags,
+				Timestamp: now,
+				Value:     value,
+			}
+
+			if err := http.sendDataPoint(metric); err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := http.flush(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *KairosDB) WriteTelnet(metrics []telegraf.Metric, u *url.URL) error {
+	// Send Data with telnet / socket communication
+	uri := fmt.Sprintf("%s:%d", u.Host, o.Port)
+	tcpAddr, _ := net.ResolveTCPAddr("tcp", uri)
+	connection, err := net.DialTCP("tcp", nil, tcpAddr)
+	if err != nil {
+		return fmt.Errorf("KairosDB: Telnet connect fail")
+	}
+	defer connection.Close()
+
+	for _, m := range metrics {
+		now := m.Time().UnixNano() / 1000000
+		tags := ToLineFormat(cleanTags(m.Tags()))
+
+		for fieldName, value := range m.Fields() {
+			switch value.(type) {
+			case int64:
+			case uint64:
+			case float64:
+			default:
+				log.Printf("D! KairosDB does not support metric value: [%s] of type [%T].\n", value, value)
+				continue
+			}
+
+			metricValue, buildError := buildValue(value)
+			if buildError != nil {
+				log.Printf("E! KairosDB: %s\n", buildError.Error())
+				continue
+			}
+
+			messageLine := fmt.Sprintf("put %s %v %s %s\n",
+				sanitize(fmt.Sprintf("%s%s%s%s", o.Prefix, m.Name(), o.Separator, fieldName)),
+				now, metricValue, tags)
+
+			_, err := connection.Write([]byte(messageLine))
+			if err != nil {
+				return fmt.Errorf("KairosDB: Telnet writing error %s", err.Error())
```

Comment:
Error messages should not start with capital letter. How about 
```suggestion
				return fmt.Errorf("writing to KairosDB using telnet failed: %v", err)
```